### DISCUSSION TITLE
How to download program with firmware

### DISCUSSION THREAD

**Discussion by Gcopper1984** [pybricks/support#816] [Q&A]
Good evening to everyone,

I managed to use pybricks (bteta) some time ago for some little project (maybe one year ago...).
In that time I managed to download program permanentely on my bricks during firmware flash with an option in the option panel.

Now I just update pybricks (beta) to the last version but I missed this option. Is it still available? Do I miss something in the new UI?

Thank you,

GC

**Comment by dlech** [comment]
With the current Pybricks Beta (firmware v3.2.x), the current program is saved when the hub is powered off and is kept in RAM after it is downloaded an run the first time. So to save a program "permanently", download and run it once, then turn the hub off (it has to be a proper shutdown using the button - you can't just removed the batteries).

**Reply by Gcopper1984** [reply]
Hi,
   ok, thank you for your reply!

Regards,

GC

**Reply by LasseD** [reply]
Hi both of you. I just want to add this info in case others run into this problem:

You have to turn the hub off by holding down the button until it turns off.
Simply removing power does not result in the program being stored on the hub (because the hub obviously needs power to write to its memory)
